Assets/_Game/Scripts/Player/PlayerWeapon.cs

```diff
@@ -119,7 +180,25 @@ public class PlayerWeapon : MonoBehaviour {
         }
       }
     }
+    else {
+      if (weaponPositions.Count > 0 && weaponPositions[0] != null) {
+        CreateWeaponAtPosition(weaponPrefab, weaponPositions[0]);
+        Weapon weaponComponent = collectedWeapons[0].GetComponent<Weapon>();
+        weaponComponent.currentWeaponLevel = 1;
+        WeaponDataLoader.Ins.LoadWeaponInfo(weaponComponent.currentWeaponId, weaponComponent.currentWeaponLevel);
```

Comment:
Why dont we use the weaponInfo returned from method WeaponDataLoader.Ins.LoadWeaponInfo?